crates/matrix-sdk/src/http_client/mod.rs

```diff
@@ -48,16 +50,47 @@ pub(crate) use native::HttpSettings;
 
 pub(crate) const DEFAULT_REQUEST_TIMEOUT: Duration = Duration::from_secs(30);
 
+#[derive(Clone, Debug)]
+struct MaybeSemaphore(Arc<Option<Semaphore>>);
+
+#[allow(dead_code)] // holding this until drop is all we are doing
```

Comment:
Is it still dead code?

Reply:
well, yes, the linter is complaining about it because the inner object is never used ... which is fine we only want to keep it for the drop ... thus it is a false-positive lint error. I left the comment but maybe that doesn't explain it well enough then?

Reply:
The comment needs some improvement indeed 😀